query_optimizer/rules/UnnestSubqueries.cpp

```diff
@@ -300,6 +302,7 @@ E::ExpressionPtr UnnestSubqueriesForNonRootLogical::eliminateOuterAttributeRefer
     std::vector<E::AttributeReferencePtr> *probe_join_attributes,
     std::vector<E::AttributeReferencePtr> *build_join_attributes,
     std::vector<E::PredicatePtr> *non_hash_join_predicates) {
+  DCHECK(expression->getExpressionType() != E::ExpressionType::kInTableQuery);
```

Comment:
Could we use `DCHECK_NE(expression->getExpressionType(), E::ExpressionType::kInTableQuery);` instead?

The style checker should suggest it whenever possible.

https://github.com/google/styleguide/blob/gh-pages/cpplint/cpplint_unittest.py#L1838

FYI, we don't need to consider the order of `checked` and `expected` values for anything but `CHECK_EQ` and `DCHECK_EQ`.